libs/SmartSync/SmartSync/Classes/Target/SFParentChildrenSyncUpTarget.m

```diff
@@ -557,9 +553,22 @@ - (BOOL)updateChildRecordInLocalStore:(SFSmartSyncSyncManager *)syncManager
                 record[kSyncTargetLocallyCreated] = @YES;
 
                 // We need a re-run
-                needReRun = true;
+                needReRun = YES;
             }
         }
+
+        // Handling remotely deleted parent
```

Comment:
That's the heart of the fix. Everything else is just refactoring to provide this method with info/objects needed.